Refine install guide (#785)
pingcap/docs-dm: en/deploy-a-dm-cluster-using-tiup.md

    config:
      log-level: info
      # rpc-timeout: "30s"
      # rpc-rate-limit: 10.0
      # rpc-rate-burst: 40
  - host: 10.0.1.18
    name: master2
    ssh_port: 22
    port: 8261
  - host: 10.0.1.19
    name: master3
    ssh_port: 22
    port: 8261

worker_servers:
  - host: 10.0.1.12
    ssh_port: 22
    port: 8262
    # deploy_dir: "/dm-deploy/dm-worker-8262"
    # log_dir: "/dm-deploy/dm-worker-8262/log"
    # numa_node: "0,1"
    # # Config is used to overwrite the `server_configs.dm-worker` values
    config:
      log-level: info
  - host: 10.0.1.19
    ssh_port: 22
    port: 8262

monitoring_servers:
  - host: 10.0.1.13
    ssh_port: 22
    port: 9090
    # deploy_dir: "/tidb-deploy/prometheus-8249"
    # data_dir: "/tidb-data/prometheus-8249"
    # log_dir: "/tidb-deploy/prometheus-8249/log"

grafana_servers:
  - host: 10.0.1.14
    port: 3000
    # deploy_dir: /tidb-deploy/grafana-3000

alertmanager_servers:
  - host: 10.0.1.15
    ssh_port: 22
    web_port: 9093
    # cluster_port: 9094
    # deploy_dir: "/tidb-deploy/alertmanager-9093"
    # data_dir: "/tidb-data/alertmanager-9093"
    # log_dir: "/tidb-deploy/alertmanager-9093/log"

> # if monitored is set, node_exporter and blackbox_exporter will be

Reviewer: 这个有测试过吗，我记得 DM 部署的时候没有 node_exporter_port 和 blackbox_exporter_port 的

Author: 模板里的呀，哥哥：https://github.com/pingcap/tiup/blob/master/embed/examples/dm/topology.example.yaml

Reviewer: 你帮忙试试有没有用

# deployed with the port specified, otherwise they are not deployed
# on the server to avoid conflict with tidb clusters
#monitored:
#  node_exporter_port: 9100
#  blackbox_exporter_port: 9115
```

> **Note:**
>
> - If you do not need to ensure high availability of the DM cluster, deploy only one DM-master node, and the number of deployed DM-worker nodes must be no less than the number of upstream MySQL/MariaDB instances to be migrated. To ensure high availability of the DM cluster, it is recommended to deploy three DM-master nodes, and the number of deployed DM-worker nodes must be greater than the number of upstream MySQL/MariaDB instances to be migrated (for example, the number of DM-worker nodes is two more than the number of upstream instances).
> 

> > - It is not recommend running too many workers on one host, each worker is recommended to be allocated at least 2 core cpu and 4 GiB memory.

Reviewer 2: ```suggestion
> - It is not recommended to run too many DM-workers on one host. Each DM-worker should be allocated at least 2 core CPU and 4 GiB memory.
```

The configuration of deploying three DM-masters, three DM-workers, and one monitoring component instance is as follows:

```yaml

> # Global variables apply to all ways of deployment. If one specific value is missing, the corresponding global variable serves as the default value.

Author: Maybe "sub sections" better than "other components " ?

    # data_dir: "/tidb-data/prometheus-8249"
    # log_dir: "/tidb-deploy/prometheus-8249/log"

grafana_servers:
  - host: 10.0.1.14
    port: 3000
    # deploy_dir: /tidb-deploy/grafana-3000

alertmanager_servers:
  - host: 10.0.1.15
    ssh_port: 22
    web_port: 9093
    # cluster_port: 9094
    # deploy_dir: "/tidb-deploy/alertmanager-9093"
    # data_dir: "/tidb-data/alertmanager-9093"
    # log_dir: "/tidb-deploy/alertmanager-9093/log"

```

> **Note:**
>
> - If you do not need to ensure high availability of the DM cluster, deploy only one DM-master node, and the number of deployed DM-worker nodes must be no less than the number of upstream MySQL/MariaDB instances to be migrated. To ensure high availability of the DM cluster, it is recommended to deploy three DM-master nodes, and the number of deployed DM-worker nodes must be greater than the number of upstream MySQL/MariaDB instances to be migrated (for example, the number of DM-worker nodes is two more than the number of upstream instances).
> 
> - It is not recommended to run too many DM-workers on one host. Each DM-worker should be allocated at least 2 core CPU and 4 GiB memory.
>

> > - The worker will be bound with an upstream database, when DM performs a full replication task, the worker exporting the full amount of data locally first, and then importing it into the downstream database. Therefore, the worker's host needs sufficient storage space (The storage path will be specified later when creating the task)

Author: 我觉得没必要，只能是资源问题，没其他原因

{{< copyable "shell-regular" >}}

```shell
tiup dm deploy ${name} ${version} ./topology.yaml -u ${ssh_user} [-p] [-i /home/root/.ssh/gcp_rsa]
```

The parameters used in this step is as follows.

|Parameter|Description|
|-|-|

> |`${name}` | The name of dm cluster, eg: dm-test|
> |`${version}` | The version of the DM cluster. You can see other supported versions by running `tiup list dm-master`.
> |`./topology.yaml`| Path of topology config file.|
> |`-u` or `--user`|Log in to the target machine through the root key to complete the cluster deployment, or you can use other users with ssh and sudo privileges to complete the deployment.|
> |`-p` or `--password`|Password of target hosts. If specified, password authentication will be used|
> |`-i` or `--identity_file`|The path of the SSH identity file. If specified, public key authentication will be used. (default "/root/.ssh/id_rsa")|

Reviewer 2: ```suggestion
|`${name}` | The name of the DM cluster, eg: dm-test|
|`${version}` | The version of the DM cluster. You can see other supported versions by running `tiup list dm-master`.
|`./topology.yaml`| The path of the topology configuration file.|
|`-u` or `--user`| Log in to the target machine as the root user or other user account with ssh and sudo privileges to complete the cluster deployment.|
|`-p` or `--password`| The password of target hosts. If specified, password authentication is used.|
|`-i` or `--identity_file`| The path of the SSH identity file. If specified, public key authentication is used (default "/root/.ssh/id_rsa").|
```

{{< copyable "shell-regular" >}}

```shell
tiup dm deploy ${name} ${version} ./topology.yaml -u ${ssh_user} [-p] [-i /home/root/.ssh/gcp_rsa]
```

The parameters used in this step is as follows.

|Parameter|Description|
|-|-|

> |`${name}` | The name of dm cluster,eg: dm-test|

Reviewer: ```suggestion
|`${name}` | The name of dm cluster, eg: dm-test|
```
